src/opnsense/service/templates/OPNsense/IDS/suricata.yaml

```diff
@@ -70,6 +70,29 @@ stats:
 
 # Configure the type of alert (and other) logging you would like.
 outputs:
+{% if helpers.exists('OPNsense.IDS.general.unix_sock') and OPNsense.IDS.general.unix_sock == '1' %}
+
+  # Extensible Event Format (nicknamed EVE) to UNIX-socket
+  - eve-log:
+      enabled: yes
+      filetype: unix_stream
+      filename: /tmp/suricata-stats.sock
```

Comment:
Well, the problem is - Suricata does not create a UNIX-socket. It connects in on an already created UNIX-socket. This socket has to be created somewhere first, before this output starts working.

Telegraf is configured to create a listening socket (check the other pull request I linked to). Telegraf is started by the user `telegraf` and `telegraf` does not have permissions to `/var/log/suricata/suricata-eve-stats.sock`. 

I am open for solutions.

Reply:
hmm, you probably have to make sure the socket is created from the suricata startup as well in some way otherwise this would likely induce a race condition. I don't have a lot of time at the moment to look into this, but glueing services together is something we usually try to avoid (question in this case probably is how to make them loosly coupled)

Reply:
if startup order matters (which I didn't try), it would be a race, but if suricata starts normally and just keeps trying to connect, it's not a huge issue. I do however still prefer to keep suricata's sockets out of /tmp in case some would like to guard the files at some point in time.

Reply:
It looks like Suricata just tries to reconnect the socket. 

Yes, I agree. World readable is probably not the best solution. However, as mentioned, this is a socket created by `telegraf` and the service Telegraf which Suricata connects to. 

The default suggestion is `source = "/var/run/suricata-stats.sock"`, but `telegraf` does not have permission to this folder. Or not that many other directories, either. 

So maybe `/var/log/telegraf/suricata-stats.sock` ..? I am very open for suggestions :)

Reply:
I think /var/run is world readable/executable by default, you just have to make sure the created socket receives proper rights when created (which would be a concern for telegraf in your case).

Reply:
Ah, yes - `telegraf` is able to read / execute. But no bueno creating/writing a listening socket in this directory; 
```
root@opnsense:/var/log/telegraf # sudo -u telegraf nc -U -l /var/run/test.sock
nc: Permission denied
```

And I do not know how I would go about changing the default permissions of this socket when created by Telegraf either (if that is what you meant with the latest comment, to make it octal notation ~`750` (?));
```
root@opnsense:/var/log/telegraf # ls -alh /tmp/suricata-stats.sock 
srwxr-xr-x  1 telegraf  wheel     0B Aug 16 13:29 /tmp/suricata-stats.sock
```

I do not know exactly how UNIX sockets works, but; stopping Telegraf and opening a shell with `sudo -u telegraf nc -U -l /tmp/suricata-stats.sock` showed streams of data from Suricata in this shell. Opening another shell, where I tried to "hop" on in to this socket, as root, with `nc -l /tmp/suricata-stats.sock` to watch the same data stream (?), the output was nothing (I do not know if this is the proper way to "tap" in to an UNIX-socket though, in regards of leaving permissions of the created socket as-is, readable/executable for others). 

The socket `/tmp/suricata-stats.sock` does not increase in size or anything either while the stream is flowing.

`/tmp` uses sticky bit, that is why the group is `wheel`.

Reply:
hmm, I think for the time being its best to keep these settings locally (in `/usr/local/etc/suricata/custom.yaml`), I currently don't have enough time to dig into how this should work and in order to add it as feature it's purpose should be consistent and clear.  The custom.yaml usually is a practical solution for more tailored setups.